lib/deploy/stepFunctions/compileStateMachines.js

```diff
@@ -99,7 +99,12 @@ module.exports = {
         let RoleArn;
         let DependsOn = [];
         let LoggingConfiguration;
-        const Tags = toTags(this.serverless.service.provider.tags);
+        let Tags;
+        if (typeof stateMachineObj.inheritGlobalTags !== 'undefined' && stateMachineObj.inheritGlobalTags === false) {
```

Comment:
would `stateMachineObj.inheritGlobalTags === false` have sufficed without the explicit `undefined` check?

Reply:
Possibly? 

I'm not as familiar with nodejs, but was trying to avoid a Key error if `inheritGlobalTags` isn't included for the state machine.

Reply:
right, I see, in this case you'll be fine, `stateMachineObj.inheritGlobalTags === false` would not match if `inheritGlobalTags` is missing

Reply:
Thanks @theburningmonk. Just pushed the update